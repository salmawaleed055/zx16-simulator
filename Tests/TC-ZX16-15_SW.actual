Loaded 18 bytes into memory
0x0000: 2271    xori x1, 17
0x0002: 6871    xori x1, 52
0x0004: 0271    xori x1, 1
0x0006: 0440    add x1, x2
0x0008: 2471    xori x1, 18
0x000A: 21B9    li a0, 16
0x000C: E623    unknown S-type
0x000E: E2A3    unknown S-type
0x0010: FFC7    Unknown opcode

--- Final State ---
x0: 0x0000
x1: 0x0036
x2: 0x0000
x3: 0x0000
x4: 0x0000
x5: 0x0000
a0: 0x0010
a1: 0x0000
PC: 0x0010
---------------------

Simulation finished.
