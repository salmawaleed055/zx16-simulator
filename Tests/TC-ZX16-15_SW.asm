# TC-ZX16-15_SW.s
# Test Objective: Validate SW (Store Word) instruction (little-endian)
# mem[a0-2] = x1[15:0]
li x1, 0x1234 # Data for SW: x1 = 0x1234
li a0, 0x1002 # Base address: a0 = 0x1002 (target 0x1000)
sw x1, -2(a0) # Store word 0x1234 from x1 at mem[0x1002 - 2] = mem[0x1000]
              # Expected: mem[0x1000]=0x34, mem[0x1001]=0x12
ecall 0x3FF