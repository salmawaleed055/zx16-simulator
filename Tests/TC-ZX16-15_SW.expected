0x0000: 2471    li x1, 18
0x0002: 23B9    li a0, 4100
0x0004: E2A3    sw x1, -2(a0)
0x0006: FFC7    ecall 0x3FF

--- Final State ---
x0: 0x0000
x1: 0x0012
x2: 0x0000
x3: 0x0000
x4: 0x0000
x5: 0x0000
a0: 0x1002
a1: 0x0000
PC: 0x0008
---------------------
Simulation finished.